test/OrchardCore.Commerce.Tests.UI/Tests/PromotionTests/PromotionBehaviourTests.cs

```diff
@@ -35,4 +36,33 @@ async Task CheckDiscountPresenceAsync(bool hasNewPrice)
                 await CheckDiscountPresenceAsync(hasNewPrice: true);
             },
             browser);
+
+    [Theory, Chrome]
+    public Task DiscountShouldProperlyAppearOnDiscountedProduct(Browser browser) =>
+        ExecuteTestAfterSetupAsync(
+            async context =>
+            {
+                // Discount should appear on discounted product's page.
+                await context.GoToRelativeUrlAsync("/testdiscountedproduct");
+                context.Get(By.CssSelector(".price-part-price-field-value > del")).Text.ShouldBe("$5.00");
+                context.Get(By.CssSelector(".price-part-price-field-value > span")).Text.ShouldBe("$3.00");
```

Comment:
I think these could be move to separate variables, since they are used multiple times. 
`.price-part-price-field-value`
`$3.00`
`$5.00`